app/utils/errors.js

```diff
@@ -0,0 +1,37 @@
+const logger = require('../logger')
+
+class CustomError extends Error {
```

Comment:
I don't like the name here. We should have a specific error essentially or each code/error meaning, for example `NotFoundError` or `OperationTimeoutError` rather than one god error. Having said that this could be used as a base class for other errors. Happy to merge as is, but this should be renamed at some point soon

Reply:
Main benefit is making the code more obvious both in implementation and reading. If you saw a HTTP 412 error would you know what that meant and whether it was the right code? I'd guess not and this is why I'd suggest using inheritance on the generic http error at least in very common cases